<launch>

	<arg name="camera_1" default="/camera_1"/>
	<arg name="camera_2" default="/camera_2"/>
  <arg name="image" default="image_raw"/>
  <arg name="transport" default="compressed"/>
  <arg name="fiducial_len" default=".1397"/> <!--0.1397 for 5.5-->
  <arg name="dictionary" default="10"/> <!--10 for 6x6, 16 for original-->
  <arg name="do_pose_estimation" default="true"/>
  <arg name="ignore_fiducials" default=""/>
  <arg name="fiducial_len_override" default="" />

<node name="tag_handler" pkg="navigation_stack" type="tag_handler.py" output="screen"/>

 <node name="camera_1" pkg="usb_cam" type="usb_cam_node" output="screen" >
   <param name="video_device" value="/dev/video0" />
   <param name="image_width" value="640" />
   <param name="image_height" value="480" />
   <param name="pixel_format" value="yuyv" />
   <param name="camera_frame_id" value="usb_cam_1" />
   <param name="io_method" value="mmap"/>
	 <param name="~camera_name" value="camera_1" />
 </node>

 <node pkg="aruco_detect" name="aruco_detect_1"
	 type="aruco_detect" respawn="false">
	 <param name="image_transport" value="$(arg transport)"/>
	 <param name="publish_images" value="true" />
	 <param name="fiducial_len" value="$(arg fiducial_len)"/>
	 <param name="dictionary" value="$(arg dictionary)"/>
	 <param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
	 <param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
	 <param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
	 <remap from="/camera/compressed"
			 to="$(arg camera_1)/$(arg image)/$(arg transport)"/>
	 <remap from="/camera_info" to="$(arg camera_1)/camera_info"/>
	<remap from="fiducial_images" to="$(arg camera_1)/fiducial_images"/>
	<remap from="fiducial_transforms" to="$(arg camera_1)/fiducial_transforms"/>
 </node>

 <node name="cam_slam_1_node" pkg="navigation_stack" type="cam_slam.py" output ="screen" args="1" />

<node name="camera_2" pkg="usb_cam" type="usb_cam_node" output="screen" >
	 <param name="video_device" value="/dev/video1" />
	 <param name="image_width" value="640" />
	 <param name="image_height" value="480" />
	 <param name="pixel_format" value="yuyv" />
	 <param name="camera_frame_id" value="usb_cam_2" />
	 <param name="io_method" value="mmap"/>
	 <param name="~camera_name" value="camera_2" />
 </node>

 <node pkg="aruco_detect" name="aruco_detect_2"
	 type="aruco_detect" respawn="false">
	 <param name="image_transport" value="$(arg transport)"/>
	 <param name="publish_images" value="true" />
	 <param name="fiducial_len" value="$(arg fiducial_len)"/>
	 <param name="dictionary" value="$(arg dictionary)"/>
	 <param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
	 <param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
	 <param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
	 <remap from="/camera/compressed"
			 to="$(arg camera_2)/$(arg image)/$(arg transport)"/>
	 <remap from="/camera_info" to="$(arg camera_2)/camera_info"/>
	<remap from="fiducial_images" to="$(arg camera_2)/fiducial_images"/>
	<remap from="fiducial_transforms" to="$(arg camera_2)/fiducial_transforms"/>
 </node>

 <node name="cam_slam_2_node" pkg="navigation_stack" type="cam_slam.py" output ="screen" args="2" />

 <node name="master_slam" pkg="navigation_stack" type="slam.py" output ="screen" args="False" />

</launch>
